relay-server/src/services/processor.rs

```diff
@@ -1229,44 +1207,47 @@ impl EnvelopeProcessorService {
         &self,
         state: &mut ProcessEnvelopeState,
         managed_envelope: &mut TypedEnvelope<Group>,
+        event: Annotated<Event>,
```

Comment:
We take ownership of the event and return it as `Some` only if it was not dropped.